### PR TITLE
Fix invited members does not reflect in offline mode

### PR DESCRIPTION
**Pull request by rezkiy37**
### Details

Actually, the changes improve how the app handles the inviting members to a workspace during the offline mode. The app previews new members like a pending state. It follows [Optimistic With Feedback Pattern B](https://github.com/Expensify/App/blob/main/contributingGuides/OFFLINE_UX.md#b---optimistic-with-feedback-pattern).
Also, once our users back online, the app sends a request to add the new members. A response includes new members with own **ids**. It is important to clear local created members (_local **ids**_) and replace them with those members come from the backend.

### Fixed Issues

$ https://github.com/Expensify/App/issues/21891
PROPOSAL: https://github.com/Expensify/App/issues/21891#issuecomment-1628546127

### Tests

1. Open the app.
2. Click on "**Settings**" (_your user avatar_).
3. Click on **"Workspaces"**. 
4. Click on any existed workspace/Click on "**New workspace**" and create a new one, if you do not have already one.
5. Click on **"Members"**.
6. Click on **"Invite"**.
7. Select any amount of new members.
8. Click on **"Next"**.
9. Click on **"Invite"**.
10. Verify that all invited users are being included in workspace members.

- [x] Verify that no errors appear in the JS console

### Offline tests

1. Open the app.
2. Click on "**Settings**" (_your user avatar_).
3. Click on **"Workspaces"**. 
4. Click on any existed workspace/Click on "**New workspace**" and create a new one, if you do not have already one.
5. Click on **"Members"**.
6. Open the "**Network**" tap of **DevTools** and select the `offline` mode.
7. Click on **"Invite"**.
8. Select any amount of new members.
9. Click on **"Next"**.
10. Click on **"Invite"**.
11. Verify that all invited users are being included in workspace members and has a pending state (_translucent_).
12. Disable the `offline` mode.
13. Verify that all invited users are being included in workspace members.

### QA Steps

Same as **"Tests"** or **"Offline tests"**.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/57314631/5edd16ad-ff68-4bec-a4dd-c54915732beb

https://github.com/Expensify/App/assets/57314631/496e45cd-0403-41e2-85f8-95d79552fc02

https://github.com/Expensify/App/assets/57314631/8c3619f6-cdec-4f3e-bbf2-8becc6d2d92f

https://github.com/Expensify/App/assets/57314631/4e1b0584-379d-40d0-80cd-d6ba4dde663e

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/57314631/bccfcb29-8ac9-4f41-a47c-6082b109ce8b

https://github.com/Expensify/App/assets/57314631/7b5ca823-813c-460e-bb46-b21963f7093f

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/57314631/22650afc-89f6-4372-bbda-2cbcdd3cc51e

https://github.com/Expensify/App/assets/57314631/43e3c915-f0df-4881-bb34-77451557bd6d

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/57314631/11effecb-46da-47f2-94ea-6a4e88ab1029

https://github.com/Expensify/App/assets/57314631/ee9d76e5-7797-452a-adff-728fb579dde2

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/57314631/b1904c8a-895f-4f32-bb4c-b5488c3b48c3

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/57314631/911f588a-a739-419e-9a65-37bb775a1dbf

https://github.com/Expensify/App/assets/57314631/46e7f8d1-9363-4147-8f90-14700f5051f7

</details>

### CODE REVIEW

File: src/libs/actions/Policy.js [hunk 1]
```diff
@@ -256,15 +257,28 @@ function createPolicyExpenseChats(policyID, invitedEmailsToAccountIDs, betas) {
...
+                avatar: UserUtils.getDefaultAvatarURL(accountID),
+                displayName: login,
+            };
+            successPersonalDetails[accountID] = null;
+        }
+
+        // If the user is not in the beta, we don't want to create any chats
+        if (!Permissions.canUsePolicyExpenseChat(betas)) {
+            return;
+        }
```
Comment (fabioh8010) [inline src/libs/actions/Policy.js:280]: Why did you move this block to inside the loop?

Comment (rezkiy37) [reply src/libs/actions/Policy.js:280]: @fabioh8010, there is a reason. The app iterates `invitedEmailsToAccountIDs`, on each iteration it should decide now: should create a new local personal details?; should remove this local personal details after?.
I moved this block to inside the loop, because I do not want to iterate `invitedEmailsToAccountIDs` twice. I am going to do it during one loop.

Comment (fabioh8010) [reply src/libs/actions/Policy.js:280]: So it's just to prevent the next call `const oldChat = ReportUtils.getChatByParticipantsAndPolicy([sessionAccountID, cleanAccountID], policyID);`, but it will push some data to `workspaceMembersChats` anyway, right?

Comment (rezkiy37) [reply src/libs/actions/Policy.js:280]: Yes, it is to prevent the logic that requires a specific `beta`. The app should always push to `workspaceMembersChats` the next fields:
```javascript
workspaceMembersChats.onyxOptimisticData.push({
     onyxMethod: Onyx.METHOD.MERGE,
     key: ONYXKEYS.PERSONAL_DETAILS_LIST,
     value: optimisticPersonalDetails,
});
workspaceMembersChats.onyxSuccessData.push({
     onyxMethod: Onyx.METHOD.MERGE,
     key: ONYXKEYS.PERSONAL_DETAILS_LIST,
     value: successPersonalDetails,
});
```

Comment (rezkiy37) [reply src/libs/actions/Policy.js:280]: It is a common logic.

**Comment by rezkiy37** [comment]
> @rezkiy37 please pull main and fix lint. `personalDetails` is replaced with `allPersonalDetails` in Policy.js

@0xmiroslav , thank you, done!

**Review by 0xmiroslav — APPROVED** [review]
LGTM!

File: src/libs/actions/Policy.js [hunk 2]
```diff
@@ -371,9 +397,25 @@ function addMembersToWorkspace(invitedEmailsToAccountIDs, welcomeNote, policyID,
             onyxMethod: Onyx.METHOD.MERGE,
             key: membersListKey,
 
-            // Convert to object with each key clearing pendingAction. We don’t
-            // need to remove the members since that will be handled by onClose of OfflineWithFeedback.
-            value: _.object(accountIDs, Array(accountIDs.length).fill({pendingAction: null, errors: null})),
+            // Convert to object with each key clearing pendingAction, when it is an existed account.
```
Comment (madmax330) [inline src/libs/actions/Policy.js:400]:
```suggestion
            // Convert to object with each key clearing pendingAction, when it is an existing account.
```

File: src/libs/actions/Policy.js [hunk 3]
```diff
@@ -371,9 +397,25 @@ function addMembersToWorkspace(invitedEmailsToAccountIDs, welcomeNote, policyID,
...
+            // Remove the object, when it is a new created account.
```
Comment (madmax330) [inline src/libs/actions/Policy.js:401]:
```suggestion
            // Remove the object, when it is a newly created account.
```

File: src/libs/actions/Policy.js [hunk 4]
```diff
@@ -371,9 +397,25 @@ function addMembersToWorkspace(invitedEmailsToAccountIDs, welcomeNote, policyID,
...
+            value: _.reduce(
+                accountIDs,
+                (memo, accountID) => {
+                    let value = null;
+                    const isExistedAccount = !_.isEmpty(allPersonalDetails[accountID]);
```
Comment (madmax330) [inline src/libs/actions/Policy.js:406]:
```suggestion
                    const accountAlreadyExists = !_.isEmpty(allPersonalDetails[accountID]);
```

**Comment by madmax330** [comment]
@0xmiroslav can you fix the conflict?

**Comment by madmax330** [comment]
Oh, my bad @rezkiy37 can you fix the conflicts please?

**Comment by rezkiy37** [comment]
Will do it tomorrow (on Monday).

File: src/libs/actions/Policy.js (hunk 2 above)
Comment (rezkiy37) [reply src/libs/actions/Policy.js]: Done - https://github.com/Expensify/App/pull/22410/commits/db2ab7aaa51849081273cc3a2eaefafd5dca5e0d

File: src/libs/actions/Policy.js (hunk 3 above)
Comment (rezkiy37) [reply src/libs/actions/Policy.js]: Done - https://github.com/Expensify/App/pull/22410/commits/db2ab7aaa51849081273cc3a2eaefafd5dca5e0d

File: src/libs/actions/Policy.js (hunk 4 above)
Comment (rezkiy37) [reply src/libs/actions/Policy.js]: Done - https://github.com/Expensify/App/pull/22410/commits/db2ab7aaa51849081273cc3a2eaefafd5dca5e0d

**Comment by rezkiy37** [comment]
> @rezkiy37 just to confirm that you fixed conflict by full reverting #22904 right?

I haven't reverted these changes https://github.com/Expensify/App/pull/22904. This is a merge commit - https://github.com/Expensify/App/pull/22410/commits/157de09f34739060c1cad6965ea041cc9804006b.

**Comment by rezkiy37** [comment]
I see they removed:
```javascript
const logins = _.map(_.keys(invitedEmailsToAccountIDs), (memberLogin) => OptionsListUtils.addSMSDomainIfPhoneNumber(memberLogin));
``` 

However, I need this data and I have it.

**Comment by madmax330** [comment]
@0xmiroslav can you please re-review?

File: src/libs/actions/Policy.js [hunk 5]
```diff
@@ -260,15 +261,28 @@ function createPolicyExpenseChats(policyID, invitedEmailsToAccountIDs, betas) {
...
         const login = OptionsListUtils.addSMSDomainIfPhoneNumber(email);
 
+        if (_.isEmpty(allPersonalDetails[accountID])) {
+            optimisticPersonalDetails[accountID] = {
+                login,
+                accountID,
+                avatar: UserUtils.getDefaultAvatarURL(accountID),
+                displayName: login,
+            };
+            successPersonalDetails[accountID] = null;
```
[the remark this answers is not in the document]
Comment (rezkiy37) [reply src/libs/actions/Policy.js]: Yep, looks like some changes were merged recently with kind of this logic. I've removed my changes to modify personal details list.

**Comment by rezkiy37** [comment]
@0xmiroslav, ready for re-review 🙂

**Comment by rezkiy37** [comment]
@0xmiroslav, just a friendly reminder 🙂

**Comment by madmax330** [comment]
Bump @0xmiroslav

**Comment by rushatgabhane** [comment]
I can review this

File: src/libs/actions/Policy.js [hunk 6]
```diff
@@ -375,9 +375,25 @@ function addMembersToWorkspace(invitedEmailsToAccountIDs, welcomeNote, policyID,
...
             key: membersListKey,
 
-            // Convert to object with each key clearing pendingAction. We don’t
-            // need to remove the members since that will be handled by onClose of OfflineWithFeedback.
-            value: _.object(accountIDs, Array(accountIDs.length).fill({pendingAction: null, errors: null})),
+            // Convert to object with each key clearing pendingAction, when it is an existing account.
+            // Remove the object, when it is a newly created account.
+            value: _.reduce(
+                accountIDs,
+                (memo, accountID) => {
```
Comment (rushatgabhane) [inline src/libs/actions/Policy.js:382]: not trying to be pedantic, but if name is `memo`, we'll have to read the implementation to understand `reduce`.

What do you think?

```suggestion
                (accountIDsWithClearedPendingAction, accountID) => {
```

**Review by rushatgabhane — CHANGES_REQUESTED** [review]

**Comment by rushatgabhane** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Review by situchan — COMMENTED** [review]
Sorry I was reviewing this already based on https://expensify.slack.com/archives/C02NK2DQWUX/p1691056044424399?thread_ts=1691055547.528309&cid=C02NK2DQWUX.
It didn't get passed 1 day yet after I got suggested to review.

**Comment by situchan** [comment]
Console error: (this also happens on main so out of scope, seems regression from https://github.com/Expensify/App/pull/23705)

Screenshot 2023-08-04 at 5 34 24 AM

**Comment by situchan** [comment]
Tried to invite both existing and non-existing users.
Confirmed optimistic accountID (non-existing user) is removed from both `personalDetailsList` and `policyMembers_xxx` upon api success.


https://github.com/Expensify/App/assets/108292595/10011ca2-bfab-4fe9-a89b-d18d947c68df

Screenshot 2023-08-04 at 5 39 56 AM

**Comment by situchan** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/108292595/10011ca2-bfab-4fe9-a89b-d18d947c68df

</details>

<details>
<summary>Mobile Web - Chrome</summary>


https://github.com/Expensify/App/assets/108292595/50d5896b-1f10-4621-9ece-98d04cdf7562



</details>

<details>
<summary>Mobile Web - Safari</summary>


https://github.com/Expensify/App/assets/108292595/40058943-c55b-40cd-a30f-b611204f62df



</details>

<details>
<summary>Desktop</summary>


https://github.com/Expensify/App/assets/108292595/517c6631-ae89-488d-9af2-47a20169a2c7



</details>

<details>
<summary>iOS</summary>


https://github.com/Expensify/App/assets/108292595/10d87a15-9836-4136-ab75-e9c41ad5a5dd


https://github.com/Expensify/App/assets/108292595/035e937d-5b23-40b2-bbfb-72876849c715



</details>

<details>
<summary>Android</summary>


https://github.com/Expensify/App/assets/108292595/d89e442c-76d5-4581-9367-453ad20dca92



</details>

**Review by situchan — APPROVED** [review]
:shipit: from me

File: src/libs/actions/Policy.js (hunk 6 above)
Comment (rezkiy37) [reply src/libs/actions/Policy.js:382]: Yeah, why not. I saw that there are many places where they call accumulator like memo, I decided to be consistent. Let's rename!

Comment (rezkiy37) [reply src/libs/actions/Policy.js]: Done - https://github.com/Expensify/App/pull/22410/commits/3b7448e2a607ddf36be4780d55ea932d6f733a2e

Comment (rushatgabhane) [reply src/libs/actions/Policy.js]: yes!

**Review by madmax330 — APPROVED** [review]
